$OpenBSD: patch-src_con_c,v 1.12 2013/11/22 19:02:50 dcoppa Exp $

commit 57822e3ed43c011ad8d28c09b31c014cd930fe9b
Author: Tony Crisci <tony@dubstepdish.com>
Date:   Fri Nov 22 10:00:49 2013 -0500

Remove-child callback skips output content cons

Every container 'above' (in the hierarchy) the workspace content should
not be closed when the last child was removed.

Add a check for output content containers and do not handle them. These
cons are at the root of the output container with dockarea siblings.
They may be run through this callback when an output is killed with
RandR.

Fixes an issue that caused content cons to become urgent when the output
is killed with RandR.

--- src/con.c.orig	Wed Aug  7 20:50:24 2013
+++ src/con.c	Fri Nov 22 19:50:19 2013
@@ -1354,8 +1354,9 @@ static void con_on_remove_child(Con *con) {
      * not be closed when the last child was removed */
     if (con->type == CT_OUTPUT ||
         con->type == CT_ROOT ||
-        con->type == CT_DOCKAREA) {
-        DLOG("not handling, type = %d\n", con->type);
+        con->type == CT_DOCKAREA ||
+        (con->parent != NULL && con->parent->type == CT_OUTPUT)) {
+        DLOG("not handling, type = %d, name = %s\n", con->type, con->name);
         return;
     }
 
